### PR TITLE
[TIMOB-23136] iOS: Deprecate TiJSCore / Kroll-Thread

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23136

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -2269,19 +2285,30 @@ iOSBuilder.prototype.initialize = function initialize() {
 	this.currentBuildManifest.encryptJS          = !!this.encryptJS
 	this.currentBuildManifest.showErrorController          = this.showErrorController
 
-	// This is default behavior for now. Move this to true in phase 2.
-	// Remove the debugHost/profilerHost check when we have debugging/profiling support with JSCore framework
-	// TIMOB-17892
-	this.currentBuildManifest.useJSCore = this.useJSCore = !this.debugHost && !this.profilerHost && (this.tiapp.ios['use-jscore-framework'] || false);
-	// Remove this check on 6.0.0
+	// Use native JSCore by default (TIMOB-23136)
+	this.currentBuildManifest.useJSCore = this.useJSCore = !this.debugHost && !this.profilerHost && (this.tiapp.ios['use-jscore-framework'] || true);
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: The check is wrong. Try:

```
this.currentBuildManifest.useJSCore = this.useJSCore = !this.debugHost && !this.profilerHost && this.tiapp.ios['use-jscore-framework'] !== false;
```

File: iphone/cli/commands/_build.js [hunk 2]
```diff
@@ -2269,19 +2285,30 @@ iOSBuilder.prototype.initialize = function initialize() {
...
+		this.logger.warn(__('run-on-main-thread no longer set in the <ios> section of the tiapp.xml. Use <property name="run-on-main-thread" type="bool">true</property> instead'));
 		this.currentBuildManifest.runOnMainThread = this.runOnMainThread = (this.tiapp.ios['run-on-main-thread'] === true);
 	} else {
 		this.currentBuildManifest.runOnMainThread = this.runOnMainThread = (this.tiapp.properties && this.tiapp.properties.hasOwnProperty('run-on-main-thread') && this.tiapp.properties['run-on-main-thread'].value || false);
 	}
 	this.currentBuildManifest.useAutoLayout = this.useAutoLayout = this.tiapp.ios && (this.tiapp.ios['use-autolayout'] === true);
 
+	// Deprecate TiJSCore and leave a warning if used anyway
+	if (!this.useJSCore) {
+		this.logger.warn(__('Titanium 7.0.0 deprecates the legacy JavaScriptCore library in favor of the built-in JavaScriptCore'.));
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: Period at the end of the line is in the wrong place:

```
... JavaScriptCore'.));
```

should be

```
... JavaScriptCore.'));
```

**Review by cb1kenobi — CHANGES_REQUESTED** [review]

**Comment by hansemannn** [comment]
@cb1kenobi

**Review by cb1kenobi — APPROVED** [review]
LGTM

**Pull request merged**